be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -279,14 +306,18 @@ Status LakePersistentIndex::merge_sstables(std::unique_ptr<sstable::Iterator> it
     return builder->Finish();
 }
 
-Status LakePersistentIndex::major_compact(int64_t min_retain_version, TxnLogPB* txn_log) {
-    if (_sstables.size() < config::lake_pk_index_sst_min_compaction_versions) {
+Status LakePersistentIndex::major_compact(const TabletMetadata& metadata, int64_t min_retain_version,
```

Comment:
read from the interface, this `major_compact` will have no access to LakePersistentIndex's member variable, so it could be utility function or static member function?

Reply:
I think we can keep it no-static now, perhaps there is a strong need for internal state later.